electrum/wallet.py

```diff
@@ -3342,6 +3342,12 @@ def restore_wallet_from_text(text, *, path, config: SimpleConfig,
     assert not storage.file_exists(), "file was created too soon! plaintext keys might have been written to disk"
     wallet.update_password(old_pw=None, new_pw=password, encrypt_storage=encrypt_file)
     wallet.synchronize()
+    if wait_for_sync:
+        if not sync_network:
+            raise Exception('sync_network is required if wait_for_sync is True')
+        wallet.start_network(sync_network)
+        while not wallet.network.is_connected:
+            time.sleep(0.1)
```

Comment:
Note that `async def restore_sync` is an async function, running on the event loop.
Hence this polling here is blocking the event loop (for the daemon, when used via RPC).
I recommend moving this out into the async context, into `restore_sync`. i.e. after `restore_wallet_from_text()` returns, implement the new logic there, outside this method, in `restore_sync`. You can poll there with e.g. `await asyncio.sleep(0.1)`, which would then not block the event loop.